configs/components/ruby.rb

```diff
@@ -108,13 +117,34 @@
     pkg.environment "LDFLAGS" => "-Wl,-rpath=/opt/puppetlabs/puppet/lib"
   end
 
+  prefix = settings[:prefix]
+
+  if platform.is_windows?
+    pkg.build_requires "pl-gdbm-#{platform.architecture}"
+    pkg.build_requires "pl-iconv-#{platform.architecture}"
+    pkg.build_requires "pl-libffi-#{platform.architecture}"
+    pkg.build_requires "pl-pdcurses-#{platform.architecture}"
+    pkg.environment "PATH" => "#{settings[:gcc_bindir]}:#{settings[:tools_root]}/bin:#{settings[:bindir]}:$$PATH"
+    pkg.environment "CYGWIN" => settings[:cygwin]
+    pkg.environment "CC" => settings[:cc]
+    pkg.environment "CXX" => settings[:cxx]
+    pkg.environment "LDFLAGS" => settings[:ldflags]
+    pkg.environment "CFLAGS" => settings[:cflags]
+    prefix = platform.convert_to_windows_path(settings[:prefix])
+
+    pkg.environment "MAKE" => platform[:make]
+
+    special_flags = "CPPFLAGS='-DFD_SETSIZE=2048' debugflags=-g"
```

Comment:
nitpick: comment pointing out why file descriptor size needs to be changed for windows? Not blocking.